pkg/hostagent/dns/dns.go

```diff
@@ -368,7 +369,7 @@ func listenAndServe(network Network, opts ServerOptions) (*dns.Server, error) {
 	var addr string
 	// always enable reply truncate for UDP
 	if network == UDP {
-		opts.TruncateReply = true
+		opts.HandlerOptions.TruncateReply = true
```

Comment:
Nit: no need for `HandlerOptions` since it is embeded.

Reply:
It is not needed, but it doesn't hurt either.